platform/conf/conf.go

```diff
@@ -504,36 +509,38 @@ func (c *Conf) addSystemdUnitV3(name, contents string, enable bool) {
 					Name:     name,
 					Contents: &contents,
 					Enabled:  &enable,
+					Mask:     &mask,
 				},
 			},
 		},
 	}
 	c.MergeV3(newConfig)
 }
 
-func (c *Conf) addSystemdUnitCloudConfig(name, contents string, enable bool) {
+func (c *Conf) addSystemdUnitCloudConfig(name, contents string, enable, mask bool) {
 	c.cloudconfig.CoreOS.Units = append(c.cloudconfig.CoreOS.Units, cci.Unit{
 		Name:    name,
 		Content: contents,
 		Enable:  enable,
+		Mask:    mask,
 	})
 }
 
-func (c *Conf) AddSystemdUnit(name, contents string, enable bool) {
+func (c *Conf) AddSystemdUnit(name, contents string, enable, mask bool) {
```

Comment:
I'm not a fan of a list of booleans in a method signature.  `true, false` isn't especially readable.  This could probably be replaced with a single argument taking an enumeration: either enable, or mask, or do neither.